geode-junit/src/main/java/org/apache/geode/test/micrometer/AbstractMeterAssert.java

```diff
@@ -38,7 +38,7 @@
    */
   AbstractMeterAssert(M meter, Class<A> selfType) {
     super(meter, selfType);
-    meterId = meter.getId();
+    meterId = meter == null ? null : meter.getId();
```

Comment:
A total nitpick but a set of parens would help readability.

Reply:
I've fixed this in a different way in the metrics session PR. My original code was an unwarranted optimization (caching the ID). Better is to defer fetching the ID until an assertion evaluates it. Given that the first step in each assertion method is to assert that the actual meter is non-null, each assertion method can safely dereference the meter to get the ID.

Reply:
It seems best to go with @demery-pivotal's changes. Whoever merges their PR last can just keep the changes from Dale's PR.